### PR TITLE
TRELLO TASK 5

### PR DESCRIPTION
**Pull request by YurekB**
made changes based on github feedback, added another test for if the votes go below zero and removed testing for invalid input

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -79,3 +79,60 @@ describe("GET Requests", () => {
...
+              article_id: 1,
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: 0,
+            })
+          );
+        });
+    });
+    test("if passed a number that will take the total to less than 0, throws an error", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:157]: What if the id given is invalid, or doesn't exist in the database? What if the inc_votes value isn't a number? What if there's no inc_votes value at all? 😱

File: app.js [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
 
+app.patch("/api/articles/:article_id", patchArticleById);
+
 app.all("/*", (req, res) => {
   res.status(404).send({ msg: "No such endpoint!" });
 });
 
 app.use((err, req, res, next) => {
-  res.status(404).send({ msg: err.msg });
+  if (err.status === 404) {
```
Comment (Rosem10) [inline app.js]: You can make this and the error handler below more dynamic and DRY-er but having res.status(err.status).send({msg: err.msg}) - then you can use the same function for both of these, AND you don't have to keep writing middleware for each custom error.

File: controllers/article.controllers.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const {
+  fetchArticleById,
+  updateArticleById,
+} = require("../models/article.models");
+
+exports.getArticleById = (req, res, next) => {
+  const id = Object.values(req.params);
```
Comment (Rosem10) [inline controllers/article.controllers.js]: You don't need Object.values here (but it works, so if you want to leave it in, I'm not mad at it). You could just go for const {article_id} = req.params to destructure it straight from the params object.

Comment (Rosem10) [reply controllers/article.controllers.js]: The above would also stop you having to use the 0 index on line 8.

File: controllers/article.controllers.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    .then((response) => {
+      res.status(200).send({ article: response });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.patchArticleById = (req, res, next) => {
+  const id = Object.values(req.params);
```
Comment (Rosem10) [inline controllers/article.controllers.js]: Same as above

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Bit more error handling to do please!

File: __tests__/app.test.js (hunk 1 above)
Comment (YurekB) [reply __tests__/app.test.js:157]: i had a test in for if inc_votes is passed something that isnt a number but when i did the pull request i got told something like we dont need that because of happy path
